pkg/minishift/util/set_driver_options_from_env.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"github.com/docker/machine/libmachine/drivers"
+	"reflect"
+	"github.com/docker/machine/libmachine/mcnflag"
+	"os"
+)
+
+func SetDriverOptionsFromEnvironment(d drivers.Driver) error {
+	supportedFlags := d.GetCreateFlags()
+
+	checkFlags := &drivers.CheckDriverOptions{
+		FlagsValues: map[string]interface{}{},
+		CreateFlags: supportedFlags,
+	}
+
+	for _, flag := range supportedFlags {
+		switch reflect.TypeOf(flag).String() {
```

Comment:
why do you switch on the string representation of the type. That's fragile, you really should switch on the type - https://golang.org/doc/effective_go.html#type_switch

But even more importantly, why the switch at all? AFAICT, the current code does not require any type switch atm.